File: C#.Net/JDI/Web/JDI Web/Selenium/Elements/Base/Clickable.cs
using System;
using JDI_Core.Interfaces.Base;
using JDI_Web.Selenium.Base;
using OpenQA.Selenium;
using OpenQA.Selenium.Interactions;

namespace JDI_Web.Selenium.Elements.Base
{
    public class Clickable : WebElement, IClickable
    {
        public Clickable() : this (null) { }
        public Clickable(By byLocator = null, IWebElement webElement = null) 
            : base(byLocator, webElement) { }
        
        protected Action<WebBaseElement> ClickAction = cl => cl.WebElement.Click();
 
        public void Click()
        {
            Actions.Click(ClickAction);
        }

        protected Action<WebBaseElement> ClickJsAction = cl => cl.JsExecutor.ExecuteScript("arguments[0].click();", cl.WebElement);

        public void ClickByXY(int x, int y)
        {
            Invoker.DoJAction($"Click on Element with coordinates (x,y) = ({x},{y})",
            el =>
                {
                    var element = WebElement;
                    new Actions(WebDriver).MoveToElement(WebElement, x, y).Click().Build().Perform();
                }
            );

        }
    }
}

File: Python/JDI/web/selenium/elements/base/clickable.py
from JDI.web.selenium.elements.base.element import Element
from JDI.core.utils.decorators import scenario


class Clickable(Element):
    def __init__(self, by_locator=None, web_element=None):
        if by_locator is not None:
            super(Clickable, self).__init__(by_locator=by_locator)
        elif web_element is not None:
            super(Clickable, self).__init__(web_element=web_element)
        else:
            super(Clickable, self).__init__()

    @scenario(action_name="Click on Element for")
    def click(self):
        self.click_action()

    def click_action(self):
        self.get_element().click()
